### PR TITLE
CB-18996 Fix Kudu config provider

### PR DESCRIPTION
**Pull request by attilabukor**
Previously, the Kudu config provider didn't actually run when creating a cluster, because instead of extending AbstractRoleConfigProvider, it was implementing CmTemplateComponentConfigProvider. This patch fixes that.

See detailed description in the commit message.

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduMasterConfigProvider.java [hunk 1]
```diff
@@ -11,19 +11,19 @@
...
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRoleConfigProvider;
 import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
 
 @Component
-public class KuduServiceConfigProvider implements CmTemplateComponentConfigProvider {
+public class KuduMasterConfigProvider extends AbstractRoleConfigProvider {
 
     @Override
-    public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
+    public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduMasterConfigProvider.java:21]: This changes the semantics. Do you need role or service level configs?

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduMasterConfigProvider.java [hunk 2]
```diff
@@ -11,19 +11,19 @@
...
         ArrayList<ApiClusterTemplateConfig> configs = Lists.newArrayList();
 
-        String cdhVersion = templateProcessor.getStackVersion();
-        if (isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERA_STACK_VERSION_7_2_11)) {
+        String cdhVersion = source.getBlueprintView().getProcessor().getStackVersion() == null ?
+            "" : source.getBlueprintView().getProcessor().getStackVersion();
+        if (isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERA_STACK_VERSION_7_2_11) && roleType == "KUDU_MASTER") {
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduMasterConfigProvider.java:26]: Assuming you indeed need a role config (and not a service-level one): The extra term `roleType == "KUDU_MASTER"` is redundant because of this below:
```
    @Override
    public List<String> getRoleTypes() {
        return List.of(KuduRoles.KUDU_MASTER);
    }
```
So this config provider will only be evaluated for that single role type anyway.

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduServiceConfigProviderTest.java [hunk 3]
```diff
@@ -20,17 +20,17 @@
 
 @ExtendWith(MockitoExtension.class)
 class KuduServiceConfigProviderTest {
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduServiceConfigProviderTest.java:22]: If the name change of the tested class is indeed justified, please also rename this UT class.

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduServiceConfigProviderTest.java [hunk 4]
```diff
@@ -39,10 +39,10 @@ public void testRangerPluginServiceConfig7211() {
...
         TemplatePreparationObject preparationObject = getTemplatePreparationObject(templateProcessor, "7.2.11");
 
-        List<ApiClusterTemplateConfig> serviceConfigs = subject.getServiceConfigs(templateProcessor, preparationObject);
+        List<ApiClusterTemplateConfig> roleConfigs = subject.getRoleConfigs("KUDU_MASTER", preparationObject);
 
-        assertEquals(1, serviceConfigs.size());
-        assertTrue(serviceConfigs.contains(config(RANGER_KUDU_PLUGIN_SERVICE_NAME, GENERATED_RANGER_SERVICE_NAME)));
+        assertEquals(1, roleConfigs.size());
+        assertTrue(roleConfigs.contains(config(RANGER_KUDU_PLUGIN_SERVICE_NAME, GENERATED_RANGER_SERVICE_NAME)));
     }
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduServiceConfigProviderTest.java:47]: As discussed in Slack, please add some test cases for `isConfigurationNeeded()` (one positive and one negative at least).

**Review by lajosrodek — CHANGES_REQUESTED** [review]

**Comment by lajosrodek** [comment]
CheckStyle error:
```
> Task :template-manager-cmtemplate:checkstyleMain
[ant:checkstyle] [ERROR] /grid/0/jenkins/workspace/workspace/cloudbreak-pull-request-builder-checkstyle-main/template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduMasterConfigProvider.java:26:101: Literal Strings should be compared using equals(), not '=='. [StringLiteralEquality]

> Task :template-manager-cmtemplate:checkstyleMain FAILED
```

**Comment by attilabukor** [comment]
Thanks for the feedback @lajosrodek, I addressed everything (hopefully)

**Review by biharitomi — APPROVED** [review]

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduMasterConfigProviderTest.java [hunk 5]
```diff
@@ -39,10 +41,20 @@ public void testRangerPluginServiceConfig7211() {
...
 
-        assertEquals(1, serviceConfigs.size());
-        assertTrue(serviceConfigs.contains(config(RANGER_KUDU_PLUGIN_SERVICE_NAME, GENERATED_RANGER_SERVICE_NAME)));
+        assertEquals(1, roleConfigs.size());
+        assertTrue(roleConfigs.contains(config(RANGER_KUDU_PLUGIN_SERVICE_NAME, GENERATED_RANGER_SERVICE_NAME)));
+        assertTrue(subject.isConfigurationNeeded(templateProcessor, preparationObject));
+    }
+
+    @Test
+    public void testCongigNotNeeded() {
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduMasterConfigProviderTest.java:52]: Nit: `Congig` -> `Config`

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM. Only one typo, but approving nevertheless. No need to correct now.
